casper/src/main/scala/io/casperlabs/casper/EquivocationDetector.scala

```diff
@@ -3,360 +3,124 @@ package io.casperlabs.casper
+              Log[F].debug(
+                s"The creator of Block ${PrettyPrinter.buildString(block)} has equivocated before}"
+              ) *> s.pure[F]
+            } else {
+              checkEquivocations(dag, block).map(
+                equivocated =>
+                  if (equivocated) {
+                    s.copy(
+                      equivocationsTracker = s.equivocationsTracker + block.getHeader.validatorPublicKey
+                    )
+                  } else {
+                    s
+                  }
               )
-              _ <- Log[F].warn(
-                    s"Ignorable equivocation: validator is $validator, creator justification is $creatorJustificationHash, latest message of creator is $latestMessageOfCreator"
-                  )
-              _ <- FunctorRaise[F, InvalidBlock].raise[Unit](IgnorableEquivocation)
-            } yield ()
+            }
+          })
+      s <- state.read
+      _ <- if (s.equivocationsTracker.contains(block.getHeader.validatorPublicKey)) {
+            FunctorRaise[F, InvalidBlock].raise[Unit](EquivocatedBlock)
+          } else {
+            Applicative[F].unit
           }
     } yield ()
 
-  private def requestedAsDependency(
-      block: Block,
-      blockBufferDependencyDag: DoublyLinkedDag[BlockHash]
-  ): Boolean =
-    blockBufferDependencyDag.parentToChildAdjacencyList.contains(block.blockHash)
-
-  private def creatorJustificationHash(block: Block): Option[BlockHash] =
-    ProtoUtil.creatorJustification(block).map(_.latestBlockHash)
-
-  // See summary of algorithm above
-  def checkNeglectedEquivocationsWithUpdate[F[_]: MonadThrowable: BlockStorage: FunctorRaise[
-    ?[_],
-    InvalidBlock
-  ]](
-      block: Block,
-      genesis: Block
-  )(implicit state: Cell[F, CasperState]): F[Unit] =
-    Monad[F].ifM(
-      isNeglectedEquivocationDetectedWithUpdate[F](
-        block,
-        genesis
-      )
-    )(FunctorRaise[F, InvalidBlock].raise[Unit](NeglectedEquivocation), Monad[F].unit)
-
-  private def isNeglectedEquivocationDetectedWithUpdate[F[_]: MonadThrowable: BlockStorage](
-      block: Block,
-      genesis: Block
-  )(implicit state: Cell[F, CasperState]): F[Boolean] =
-    for {
-      s <- Cell[F, CasperState].read
-      neglectedEquivocationDetected <- s.equivocationsTracker.toList.existsM { equivocationRecord =>
-                                        updateEquivocationsTracker[F](
-                                          block,
-                                          equivocationRecord,
-                                          genesis
-                                        )
-                                      }
-    } yield neglectedEquivocationDetected
-
   /**
-    * If an equivocation is detected, it is added to the equivocationDetectedBlockHashes, which keeps track
-    * of the block hashes that correspond to the blocks from which an equivocation can be justified.
+    * check whether block equivocate
```

Comment:
```suggestion
    * Check whether block equivocates
```
and it's probably a bad wording since, as @selfdual-brain keeps repeating, equivocations are always pairs. Maybe we should say sth like "Check whether a block creates equivocations".

Reply:
Done.